File: core/core-domain/src/main/java/com/coffee/gu/order/OrderRepository.java
package com.coffee.gu.order;

import com.coffee.gu.enums.OrderState;

import java.util.List;
import java.util.Optional;

public interface OrderRepository {
    Order create(Order order);
    Order save(Order order);
    List<Order> getByPrincipalKey(String principalKey);
    Optional<Order> findByOrderKey(String orderKey);
    Optional<Order> findByOrderKey(String orderKey, OrderState state);
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/order/OrderRepository.kt
package com.coffee.gu.order

import com.coffee.gu.enums.OrderState
import java.util.Optional

interface OrderRepository {
    fun create(order: Order): Order
    fun save(order: Order): Order
    fun getByPrincipalKey(principalKey: String): List<Order>
    fun findByOrderKey(orderKey: String): Order
    fun findByOrderKey(orderKey: String, state: OrderState): Order
}
